app/src/main/kotlin/io/imhungry/notifications/NotificationHelper.kt

```diff
@@ -0,0 +1,97 @@
+package io.imhungry.notifications
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import io.imhungry.R
+
+class NotificationHelper(private val context: Context) {
+
+    /**
+     *  Unfortunately, NotificationCompat does actually have an ENUM behind the scenes.
+     *  This simplifies the function arguments in this class.
+     */
+    enum class Priority constructor(internal val value: Int) {
+        MIN(NotificationCompat.PRIORITY_MIN),
+        LOW(NotificationCompat.PRIORITY_LOW),
+        DEFAULT(NotificationCompat.PRIORITY_DEFAULT),
+        HIGH(NotificationCompat.PRIORITY_HIGH),
+        MAX(NotificationCompat.PRIORITY_MAX)
+    }
```

Comment:
I am really torn on the existence of this enum, in general. I personally like seeing the type-safe approach like this, but it defeats the purpose as to why Google used static variables in the first place. I am interested to hear input from others on this, or more info from you. @tiemonl

Reply:
I went with an enum because it's a helper class.  I want it to be straight forward what priority argument embodies. This way, the IDE will auto-complete the argument and give you options. 

I'm open to changes this just seemed to make the most sense for this case.

Any idea why Google used static variables here?

Reply:
I completely agree with your motivation, and in so many ways I wish Google had used enums to start. The original reason was performance and overhead. Static variables are available instantly at runtime, while Enums are entire objects that need to be created.

If the enum was offering anything beyond slight type safety, or was being used in more than a few instances, I would agree with this. The problem is it doesn't, in this instance. I think you should just rename parameters that use the Priority class to be as descriptive as possible, and possibly have a link to documentation about priority.

Reply:
Decided to keep. The enum really won't get used by the system much. Should only be once when a notification is created.